feat(bzlmod): add python.override APIs (#2222)
bazelbuild/rules_python: docs/toolchains.md

{gh-path}`test_current_py_toolchain <tests/load_from_macro/BUILD.bazel>` target
for an example.

### Overriding toolchain defaults and adding more versions

One can perform various overrides for the registered toolchains from the root
module. For example, the following use cases would be supported using the
existing attributes:

* Limiting the available toolchains for the entire `bzlmod` transitive graph
  via {attr}`python.override.available_python_versions`.

> * Setting particular `X.Y.Z` python versions when modules request `X.Y` version

Reviewer: ```suggestion
* Setting particular `X.Y.Z` Python versions when modules request `X.Y` version
```

  via {attr}`python.override.minor_mapping`.

> * Adding custom {attr}`python.single_version_platform_override.coverage_tool`.

Reviewer: ```suggestion
* Per-version control of the coverage tool used using
  {attr}`python.single_version_platform_override.coverage_tool`.
```

> * Adding new python versions via {bzl:obj}`python.single_version_override` or

Reviewer: ```suggestion
* Adding additional Python versions via {bzl:obj}`python.single_version_override` or
```
